controller/controller_test.go

```diff
@@ -543,6 +663,32 @@ func TestControllerMutation(t *testing.T) {
 			},
 			wantErr: true,
 		},
+		{
+			desc: "request dual-stack loadbalancer IPs via custom annotation",
+			in: &v1.Service{
+				ObjectMeta: metav1.ObjectMeta{
+					Annotations: map[string]string{
+						annotationLoadBalancerIPs: "1.2.3.0,1000::",
+					},
+				},
+				Spec: v1.ServiceSpec{
+					ClusterIPs: []string{"1.2.3.4", "3000::1"},
+					Type:       "LoadBalancer",
+				},
+			},
+			want: &v1.Service{
+				ObjectMeta: metav1.ObjectMeta{
+					Annotations: map[string]string{
+						annotationLoadBalancerIPs: "1.2.3.0,1000::",
+					},
+				},
+				Spec: v1.ServiceSpec{
+					Type:       "LoadBalancer",
+					ClusterIPs: []string{"1.2.3.4", "3000::1"},
+				},
+				Status: statusAssigned([]string{"1.2.3.0", "1000::"}),
+			},
+		},
```

Comment:
what if u have only one IP and we are in dual-stack mode would that flag an error ? can we add test coverage for this case too ?

Reply:
yes, it must throw an error as per the existing LoadBalancerIP unit test. added a test to cover that too.